packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -219,3 +222,46 @@ export const typeMatchesSomeSpecifier = (
   program: ts.Program,
 ): boolean =>
   specifiers.some(specifier => typeMatchesSpecifier(type, specifier, program));
+
+export function valueMatchesSpecifier(
+  node: TSESTree.Node,
+  specifier: TypeOrValueSpecifier,
+  scopeManager: ScopeManager,
+): boolean {
+  if (
+    node.type === AST_NODE_TYPES.Identifier ||
+    node.type === AST_NODE_TYPES.JSXIdentifier
+  ) {
+    if (typeof specifier === 'string') {
+      return node.name === specifier;
+    }
+
+    if (specifier.from === 'package') {
+      const variable = scopeManager.variables.find(v => v.name === node.name);
+      const targetNode = variable?.defs[0].parent;
+      if (targetNode?.type !== AST_NODE_TYPES.ImportDeclaration) {
```

Comment:
[Bug] What if the variable comes from a dynamic package import?

```ts
declare const fs: typeof import("fs");

fs.exists;
// ~~~~~~
```

Using the scope manager to get to the variable's local definition is one step. You might be able to get this to work with more AST+scope work. Or maybe it'd be easier to use TypeScript types and/or symbols? Or some combination of them? YMMV.

Reply:
I’ve added support for the dynamic import case as well.[1578137](https://github.com/typescript-eslint/typescript-eslint/pull/10670/commits/15781373b072fcfc641fe28abd7628ea4676a31f)